src/tool/xmeta/main.cpp

```diff
@@ -6,25 +6,50 @@
 
 #include "xmeta_idl_reader.h"
 #include "ast_to_st_listener.h"
+#include "xlang_model_walker.h"
+#include "xmeta_emit.h"
 
 #if defined(_WIN32)
 #include <Windows.h>
+#include <mscoree.h>
+#include <cor.h>
+#include <winrt/base.h>
 #endif
 
 #pragma execution_character_set("utf-8")
 
 using namespace antlr4;
+using namespace winrt;
+using namespace xlang::xmeta;
+
+std::string remove_extension(const std::string& filename) {
```

Comment:
Prefer to do these with string_view. It's not typically needed to allocate a whole new string for this sort of thing.